modules/social_features/social_core/social_core.install

```diff
@@ -358,3 +359,54 @@ function social_core_update_123004(): void {
     }
   }
 }
+
+/**
+ * Install the secret file system.
+ */
+function social_core_update_130001() : void {
+  \Drupal::service('module_installer')->install(['secret_file_system']);
+}
+
+/**
+ * Migrate private file system to secret file system.
+ */
+function social_core_update_130002() : void {
+  $config_ids = [
+    'field.storage.node.field_book_image',
+    'field.storage.comment.field_comment_files',
+    'field.storage.block_content.field_hero_image',
+    'field.storage.node.field_event_image',
+    'field.storage.paragraph.field_featured_item_image',
+    'field.storage.group.field_group_image',
+    'field.storage.paragraph.field_hero_image',
+    'field.storage.paragraph.field_hero_small_image',
+    'field.storage.node.field_files',
+    'field.storage.node.field_page_image',
+    'field.storage.post.field_post_image',
+    'field.storage.profile.field_profile_banner_image',
+    'field.storage.profile.field_profile_image',
+    'field.storage.node.field_topic_image',
+  ];
```

Comment:
There's a few open questions here:

1) Do we want to hard-code this list or look up all file fields and switch them over
2) In case of a hard-coded list, should that list be in social_core or spread over to individual modules?
3) In case of a hard-coded list, is this list correct? Otherwise, should there be exceptions?

Reply:
> 1. Do we want to hard-code this list or look up all file fields and switch them over

If we look up the file fields we will automatically updates the ones people built on top of Open Social too, and that is pretty neat imho.
That also applies to us too, so we don't have to do additional work down the line for it :)

As an example the `social_course` module has several file fields too. We would be auto updating them too: https://git.drupalcode.org/project/social_course/-/blob/5.1.x/config/install/field.field.paragraph.attachment.field_files.yml?ref_type=heads.

That being said, I can also see why it might be nicer to do this per individual module. Because that gives more control to only include this for fields Open Social provides...

So maybe the cleanest is if we go for the latter. Gives the most control to the site owners.

> 2. In case of a hard-coded list, should that list be in social_core or spread over to individual modules?

For me it's fine if it lives in `social_core`. But for other modules outside of the scope we'll have to write separate update hook.s

> 3. In case of a hard-coded list, is this list correct? Otherwise, should there be exceptions?

I think we're missing:
- `field.storage.media.field_media_file`